packages/dao/contracts/Api3Template.sol

```diff
@@ -59,46 +64,55 @@ contract Api3Template is Api3BaseTemplate {
             address(dao),
             address(acl),
             address(_api3Pool),
-            address(voting),
-            address(agent)
+            address(mainVoting),
+            address(secondaryVoting),
+            address(mainAgent),
+            address(secondaryAgent)
         );
     }
 
     function _setupApps(
         Kernel _dao,
         ACL _acl,
         MiniMeToken _api3Pool,
-        uint64[3] memory _votingSettings,
+        uint64[3] memory _mainVotingSettings,
+        uint64[3] memory _secondaryVotingSettings,
         address _permissionManager
     )
     internal
-    returns (Api3Voting, Agent)
+    returns (Api3Voting, Api3Voting, Agent, Agent)
     {
-        Agent agent = _installDefaultAgentApp(_dao);
-        Api3Voting voting = _installVotingApp(_dao, _api3Pool, _votingSettings);
+        (Agent mainAgent, Agent secondaryAgent) = _installDefaultAgentApps(_dao);
+        (Api3Voting mainVoting, Api3Voting secondaryVoting) = _installVotingApps(_dao, _api3Pool, _mainVotingSettings, _secondaryVotingSettings);
 
         _setupPermissions(
             _acl,
-            agent,
-            voting,
+            mainAgent,
+            mainVoting,
+            secondaryAgent,
+            secondaryVoting,
             _permissionManager
         );
 
-        return (voting, agent);
+        return (mainVoting, secondaryVoting, mainAgent, secondaryAgent);
     }
 
     function _setupPermissions(
         ACL _acl,
-        Agent _agent,
-        Api3Voting _voting,
+        Agent _mainAgent,
+        Api3Voting _mainVoting,
+        Agent _secondaryAgent,
+        Api3Voting _secondaryVoting,
         address _permissionManager
     )
     internal
     {
-        _createAgentPermissions(_acl, _agent, _voting, _permissionManager);
-        _createVaultPermissions(_acl, Vault(_agent), _voting, _permissionManager);
+        _createAgentPermissions(_acl, _mainAgent, _mainVoting, _permissionManager);
+        _createAgentPermissions(_acl, _secondaryAgent, _secondaryVoting, _permissionManager);
+        _createVaultPermissions(_acl, Vault(_mainAgent), _mainVoting, _permissionManager);
         _createEvmScriptsRegistryPermissions(_acl, _permissionManager, _permissionManager);
```

Comment:
You put the `_permissionManager` as the grantee of this role ? It' usually a voting app. Otherwise the `_permissionManager ` can grant himself the ability to execute custom scripts which is dangerous.

Now, given that this `_permissionManager` controls all the roles, he can do whatever he wants anyhow. That's what I find it weird that the `_permissionManager` is not actually a voting app.

Reply:
I would agree, but this permission thing was here before us, I think we should ask Burak about it, really good comment actually..

Reply:
Again, this is just a contract dOrg copy-pasted from an earlier project https://github.com/Lid-Protocol/lid-dao-contracts/ (unfortunately has piss-poor documentation) It was only supposed to be a starting point and very well may not be safe.

Reply:
See comment above.